sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shims/GpuCastShims.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /*** spark-rapids-shim-json-lines
+{"spark": "340"}
+spark-rapids-shim-json-lines ***/
+
+package com.nvidia.spark.rapids.shims
+
+import ai.rapids.cudf.{ColumnVector, ColumnView, DType}
+import com.nvidia.spark.rapids.jni.CastStrings
+
+object GpuCastShims {
+  def CastDecimalToString(decimalInput: ColumnView, ansiMode: Boolean): ColumnVector = {
+    if (ansiMode) {
+      decimalInput.castTo(DType.STRING)
+    } else {
+      CastStrings.fromDecimal(decimalInput)
```

Comment:
nit: Can we put in a comment here about what these APIs correspond to?  Just to make it simpler to match up the CPU code.

```
if (ansiMode) {
  // This is equivalent to https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#toPlainString--
  decimalInput.castTo(DType.STRING)
} else {
  // This is equivalent to https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#toString--
  CastStrings.fromDecimal(decimalInput)
}
```